src/framework/graphics/picker.js

```diff
@@ -194,73 +190,12 @@ class Picker {
-
-            DebugGraphics.pushGpuMarker(device, 'PICKER');
-
-            const renderer = this.renderer;
-            const srcLayers = scene.layers.layerList;
-            const subLayerEnabled = scene.layers.subLayerEnabled;
-            const isTransparent = scene.layers.subLayerList;
-
-            const pickColorId = device.scope.resolve('uColor');
-            const pickColor = this.pickColor;
-
-            for (let i = 0; i < srcLayers.length; i++) {
-                const srcLayer = srcLayers[i];
-
-                // skip the layer if it does not match the provided ones
-                if (layers && layers.indexOf(srcLayer) < 0) {
-                    continue;
-                }
-
-                if (srcLayer.enabled && subLayerEnabled[i]) {
-
-                    // if the layer is rendered by the camera
-                    const layerCamId = srcLayer.cameras.indexOf(camera);
-                    if (layerCamId >= 0) {
-
-                        // if the layer clears the depth
-                        if (srcLayer._clearDepthBuffer) {
-                            renderer.clear(camera.camera, false, true, false);
-                        }
-
-                        const culledInstances = srcLayer.getCulledInstances(camera.camera);
-                        const meshInstances = isTransparent[i] ? culledInstances.transparent : culledInstances.opaque;
-
-                        // only need mesh instances with a pick flag
-                        for (let j = 0; j < meshInstances.length; j++) {
-                            const meshInstance = meshInstances[j];
-                            if (meshInstance.pick) {
-                                tempMeshInstances.push(meshInstance);
-
-                                // keep the index -> meshInstance index mapping
-                                this.mapping.set(meshInstance.id, meshInstance);
-                            }
-                        }
-
-                        const lights = [[], [], []];
-                        renderer.renderForward(camera.camera, tempMeshInstances, lights, SHADER_PICK, (meshInstance) => {
-                            const miId = meshInstance.id;
-                            pickColor[0] = ((miId >> 16) & 0xff) / 255;
-                            pickColor[1] = ((miId >> 8) & 0xff) / 255;
-                            pickColor[2] = (miId & 0xff) / 255;
-                            pickColor[3] = ((miId >> 24) & 0xff) / 255;
-                            pickColorId.setValue(pickColor);
-                            device.setBlendState(BlendState.NOBLEND);
-                        });
-
-                        tempMeshInstances.length = 0;
-                    }
-                }
-            }
 
-            DebugGraphics.popGpuMarker(device);
-        });
 
-        DebugHelper.setName(renderPass, `RenderPass-Picker`);
+
+
+        /////////// !!!!!!!! todo  - this should be done once and RT resized when needed
```

Comment:
Uhm....maybe turn down the volume! 🤣

Reply:
haha yeah sorry I missed it even when it was so obvious! :)